src/azure-cli/azure/cli/command_modules/acs/managed_cluster_decorator.py

```diff
@@ -6265,6 +6268,27 @@ def update_identity(self, mc: ManagedCluster) -> ManagedCluster:
                     user_assigned_identities=user_assigned_identity
                 )
             mc.identity = identity
+        # current and goal are both userassigned, but the identity is different, update user assigned identity
+        if (
+            current_identity_type == goal_identity_type and
+            goal_identity_type == "userassigned" and
+            current_user_assigned_identity != assign_identity
+        ):
+            msg = (
+                "Your cluster is already using userassigned managed identity, current control plane identity is {},"
+                "and you are going to update the cluster identity to {}.\n"
+                "Are you sure you want to perform this operation?"
+            ).format(current_user_assigned_identity, assign_identity)
+            if not self.context.get_yes() and not prompt_y_n(msg, default="n"):
+                raise DecoratorEarlyExitException
+            user_assigned_identity = {
+                assign_identity: self.models.ManagedServiceIdentityUserAssignedIdentitiesValue()
+            }
+            identity = self.models.ManagedClusterIdentity(
+                type="UserAssigned",
+                user_assigned_identities=user_assigned_identity
+            )
+            mc.identity = identity
```

Comment:
It seems to me 6284\~6291 is idendical to 6263\~6270, I guess you just to to add the additional check before 6263?

Reply:
Also it would be great if you could help add a new live test case in test_aks_commands.py

Reply:
for1), yes, I wrote the code in this way for easy reading, I can move these code to previous `if` statement if you think it's better.
for 2), sure, will add it.

Reply:
just want to get your confirmation on 1)

Reply:
yeah, I think shorter is better :)